text_collector_examples/nvme_metrics.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+set -eu
+
+# Dependencies: nvme-cli, jq (packages)
+# Based on code from
+# - https://github.com/prometheus/node_exporter/blob/master/text_collector_examples/smartmon.sh
+# - https://github.com/vorlon/check_nvme/blob/master/check_nvme.sh
+# Author: Henk <henk@wearespindle.com>
+
+output_format_awk="$(
+  cat <<'OUTPUTAWK'
+BEGIN { v = "" }
+v != $1 {
+  print "# HELP nvme_" $1 " SMART metric " $1;
+  print "# TYPE nvme_" $1 " gauge";
+  v = $1
+}
+{print "nvme_" $0}
+OUTPUTAWK
+)"
+
+format_output() {
+  sort | awk -F'{' "${output_format_awk}"
+}
+
+# Get the nvme-cli version
+nvme_version="$(nvme version | awk '$1 == "nvme" {print $3}')"
+echo "nvmecli{version=\"${nvme_version}\"} 1" | format_output
+
+# Cut device name to /dev/nvmeX
+device_list="$(nvme list | awk '/^\/dev/{print $1}')"
+
+# Loop through the NVMe devices
+for device in ${device_list}; do
+  json_check="$(nvme smart-log -o json "${device}")"
+  disk="$(echo "${device}" | cut -c6-10)"
+
+  # The temperature value in JSON is in Kelvin, we want Celsius
+  value_temperature="$(echo "$json_check" | jq '.temperature - 273')"
+  echo "temperature_celcius{disk=\"${disk}\"} ${value_temperature}"
```

Comment:
One last nit pick.  Let's use the label name `device`, this way it matches with the `node_disk_...` label. :smile:

Reply:
👍